pkg/uhttp/wrapper.go

```diff
@@ -213,6 +215,20 @@ func WithResponse(response interface{}) DoOption {
 	}
 }
 
+func retryValueToTime(value string, now time.Time) (time.Time, error) {
```

Comment:
We should probably use ExtractRateLimitData in `pkg/ratelimit/http.go`, since it handles lots of different rate limit headers. If that doesn't do what we want, it should be modified to improve its behavior.